File: backend/src/modules/commons/entities/user.entity.ts
import { User } from "@prisma/client";

export class UserEntity {
  id: string;
  name: string;
  email: string;
  pictureUrl?: string;
  createdAt: Date;

  constructor(
    id: string,
    name: string,
    email: string,
    pictureUrl?: string ,
    createdAt: Date = new Date()
  ) {
    this.id = id;
    this.name = name;
    this.email = email;
    this.pictureUrl = pictureUrl;
    this.createdAt = createdAt;
  }

  static fromPrisma(user: User): UserEntity {
    return new UserEntity(user.id,user.name, user.email, user.picture_url, user.createdAt);
  }
}

File: frontend/falatu_mobile/lib/commons/core/domain/entities/user_entity.dart
import "package:equatable/equatable.dart";

class UserEntity extends Equatable {
  final String id;
  final String name;
  final String email;
  final String? pictureUrl;

  const UserEntity(
      {required this.id,
      required this.name,
      required this.email,
      required this.pictureUrl});

  @override
  List<Object?> get props => [id, name, email, pictureUrl];
}
